src/Workspaces/CoreTest/FindReferencesTests.cs

```diff
@@ -322,7 +323,12 @@ public System.Uri Get()
 
             var references = (await SymbolFinder.FindReferencesAsync(interfaceMethod, solution)).ToList();
             Assert.Equal(2, references.Count);
-            Assert.True(references.Any(r => r.DefinitionAndProjectId.ProjectId == desktopProject.Id));
+
```

Comment:
Add tests to SolutionTests.cs that cover all cases handled by GetExactProjectId (modules e.g.)